plugins/outputs/otlp/otlp_test.go

```diff
@@ -0,0 +1,259 @@
+package otlp
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	metricsService "go.opentelemetry.io/proto/otlp/collector/metrics/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/status"
+)
+
+type metricServiceServer struct {
+	status *status.Status
+	metricsService.UnimplementedMetricsServiceServer
+	reqs []*metricsService.ExportMetricsServiceRequest
+}
+
+func (s *metricServiceServer) Export(ctx context.Context, req *metricsService.ExportMetricsServiceRequest) (*metricsService.ExportMetricsServiceResponse, error) {
+	var emptyValue = metricsService.ExportMetricsServiceResponse{}
+	s.reqs = append(s.reqs, req)
+
+	if s.status == nil {
+		return &emptyValue, nil
+	}
+
+	return nil, s.status.Err()
+}
+
+func (s *metricServiceServer) clear() {
+	s.reqs = []*metricsService.ExportMetricsServiceRequest{}
+}
+
+func newLocalListener() net.Listener {
+	l, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		if l, err = net.Listen("tcp6", "[::1]:0"); err != nil {
+			panic(fmt.Sprintf("httptest: failed to listen on a port: %v", err))
+		}
+	}
+	return l
+}
+
+var (
+	listener          net.Listener
+	mockMetricsServer metricServiceServer
+)
+
+func TestMain(m *testing.M) {
+	listener = newLocalListener()
+	grpcServer := grpc.NewServer()
+	mockMetricsServer = metricServiceServer{
+		status: nil,
+	}
+	metricsService.RegisterMetricsServiceServer(grpcServer, &mockMetricsServer)
+	go func() {
+		_ = grpcServer.Serve(listener)
+	}()
+	defer grpcServer.Stop()
+	os.Exit(m.Run())
+}
```

Comment:
Can you please create a listener in each test separately to prevent side-effects between the tests. You might also want to take a look at https://pkg.go.dev/golang.org/x/net/nettest for creating the listener...

Reply:
Done. Thanks for pointing me to `nettest`!